File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        
        if(s1.length()>s2.length())return false;
        int[] data = new int[26];
        int[] test = new int[26];
        
        for(char c : s1.toCharArray())data[c-'a']++;
        
        for(int i=0;i<s1.length();i++)test[s2.charAt(i)-'a']++;        
        int n=s1.length(); 

        for(int i=0;i<s2.length()-n;i++){
            if(equalsString(test, data))return true;
            test[s2.charAt(i+n)-'a']++;     
            test[s2.charAt(i)-'a']--;            
        }
        return equalsString(test, data);
    }
    private boolean equalsString(int[] s1, int[] s2){
        
        for(int i=0;i<26;i++)
            if(s1[i]!=s2[i])return false;        
        return true;
    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        int n = s1.size();
        int m = s2.size();
        for(int i = 0; i <= m - n; i++){
            string target = s2.substr(i,n);
            int array[26] = {0};
            
            for(int j = 0; j < n; j++){
                array[s1[j] - 'a']++;
                array[target[j] - 'a']--;
            }
            bool flag = true;
            for(int j = 0; j < 26; j++){
                if(array[j] != 0) {
                flag = false;
                break;
                }
            }
            if(flag) return true;
        }
        return false;
    }
};